components/script/dom/webidls/GamepadHapticActuator.webidl

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+// https://w3c.github.io/gamepad/#gamepadhapticactuator-interface
+[Exposed=Window, Pref="dom.gamepad.enabled"]
+interface GamepadHapticActuator {
+  /* [SameObject] */ readonly attribute /* FrozenArray<GamepadHapticEffectType> */ any effects;
+  Promise<GamepadHapticsResult> playEffect(
```

Comment:
Let's set this up as mentioned in zullip, just so that we don't have to check the realm manually in the method.